### PR TITLE
{Docs} Documentation update for debugging this repository using VS Code

### PR DESCRIPTION
**Pull request by niswitze**
**Description**  
Updates for setting up a development environment to debug this project in VS Code. Removing the section on using the property python.pythonPath in settings.json as this property has become deprecated in the current Python extension for VS Code.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Docs

**Comment by fengzhou-msft** [comment]
@niswitze should we also update `launch.json`, change to `"pythonPath": "${command:python.interpreterPath}"`?
https://github.com/Azure/azure-cli/blob/d8973251b381f29c530c0f58afb7d5ae650351e2/.vscode/launch.json#L8

**Comment by niswitze** [comment]
> @niswitze should we also update `launch.json`, change to `"pythonPath": "${command:python.interpreterPath}"`?
> https://github.com/Azure/azure-cli/blob/d8973251b381f29c530c0f58afb7d5ae650351e2/.vscode/launch.json#L8

I would say let's leave that up to the user. They have a few options for setting this value and we should probably leave it to whichever method they prefer.

**Review by fengzhou-msft — APPROVED** [review]

File: doc/configuring_your_machine.md
Comment (fengzhou-msft) [inline doc/configuring_your_machine.md:48]:
```suggestion
    - You can also use your workspace or use settings.json, or launch.json file to specify the interpreter's path. This approach is detailed in the Environments section of the Python VS Code extension [documentation](https://code.visualstudio.com/docs/python/environments#_choose-a-debugging-environment).
```

**Comment by jiasli** [comment]
> @niswitze should we also update `launch.json`, change to `"pythonPath": "${command:python.interpreterPath}"`?
> https://github.com/Azure/azure-cli/blob/d8973251b381f29c530c0f58afb7d5ae650351e2/.vscode/launch.json#L8

`${config:python.pythonPath}` should now be `${command:python.interpreterPath}`. There is an error in VSCode Python's doc (https://github.com/microsoft/vscode-python/issues/13380).

**Comment by jiasli** [comment]
Perhaps we should put this on hold, given that the doc from VSCode Python is still self-conflicting (https://github.com/microsoft/vscode-python/issues/13380).

https://code.visualstudio.com/docs/python/environments#_choose-a-debugging-environment

> `python.pythonPath` setting in the workspace `settings.json`

https://code.visualstudio.com/docs/python/debugging#_python

> `python`: Full path that points to the Python interpreter to be used for debugging.

File: doc/configuring_your_machine.md
Comment (jiasli) [inline doc/configuring_your_machine.md:53]: Perhaps we can add the example back?

File: doc/configuring_your_machine.md
Comment (fengzhou-msft) [inline doc/configuring_your_machine.md:48]:
```suggestion
    - You can also use your workspace or user `settings.json`, or `launch.json` file to specify the interpreter's path. This approach is detailed in the Environments section of the Python VS Code extension [documentation](https://code.visualstudio.com/docs/python/environments#_choose-a-debugging-environment). An example in `settings.json`:
    ```
      {
          "python.pythonPath": "C:\\repo\\env\\Scripts\\python.exe"
      }
      ```
    or in `launch.json`
    ```
      {
          "pythonPath": "C:\\repo\\env\\Scripts\\python.exe"
      }
      ```
```

File: doc/configuring_your_machine.md
Comment (fengzhou-msft) [inline doc/configuring_your_machine.md:56]:
```suggestion
      or in `launch.json`
    ```
      {
```

**Review by jiasli — APPROVED** [review]

**Pull request merged**
